gateway/src/project.rs

```diff
@@ -687,9 +689,33 @@ where
     type Next = ProjectStarting;
     type Error = ProjectError;
 
-    async fn next(self, _ctx: &Ctx) -> Result<Self::Next, Self::Error> {
+    async fn next(self, ctx: &Ctx) -> Result<Self::Next, Self::Error> {
+        let container_id = self
+            .container
+            .id
+            .as_deref()
+            .expect("container should have ID");
+
+        // Filter and count the `start` events for this project in the last 15 minutes
+        let start_event_count = ctx
+            .docker()
+            .events(Some(EventsOptions::<String> {
+                since: Some(
+                    (chrono::offset::Utc::now() - chrono::Duration::minutes(15)).to_string(),
+                ),
+                until: Some(chrono::offset::Utc::now().to_string()),
+                filters: HashMap::from([
+                    ("container".to_string(), vec![container_id.to_string()]),
+                    ("event".to_string(), vec!["start".to_string()]),
+                ]),
+            }))
+            .count()
```

Comment:
A possible pointer for why this is not working is that this is a stream of `Result`s. So you may be just counting a single error before the stream terminates?

Reply:
This was the case, and replacing it with `try_collect` showed me that I was using the wrong time format for the filters. :man_facepalming: Thanks!